This does not exist in normal shaderc and was modified[1] by PCSX2
for use in their CI/appimage making it impossible to use system's
(despite they do not bundle it). So not much choices but to remove
usage.

Partial (for simplification) revert of [2], hopefully temporary.

[1] https://github.com/PCSX2/pcsx2/commit/1cd4ba269854646
[2] https://github.com/PCSX2/pcsx2/commit/fbfdf1200d27260
--- a/pcsx2/GS/Renderers/Vulkan/VKShaderCache.cpp
+++ b/pcsx2/GS/Renderers/Vulkan/VKShaderCache.cpp
@@ -113,6 +113,4 @@
 	{
 		options.SetGenerateDebugInfo();
-		if (GSDeviceVK::GetInstance()->GetOptionalExtensions().vk_khr_shader_non_semantic_info)
-			options.SetEmitNonSemanticDebugInfo();
 
 		options.SetOptimizationLevel(shaderc_optimization_level_zero);
